akka-javasdk/src/main/scala/akka/javasdk/impl/agent/FunctionTools.scala

```diff
@@ -49,7 +50,7 @@ object FunctionTools {
           override def types: Array[Type] =
             method.getGenericParameterTypes
 
-          override def invoke(obj: Any, args: Array[Any]): Any = {
```

Comment:
@aludwiko, I know I said I would preserve this change, but it's not possible. 

The main reason is that for some tool calling, we won't have an instance at all. For example, when calling a Component, we will have to use the class to build a method ref and then call it as we do inside the component client.

We also want to add tools by passing the .class only. When it comes the moment to initialize it we will need to either use a default constructor or use the DependencyProvider to create an instance. It will be better to have this done inside the `FunctionToolInvoker` to shield away the specifics of object initialization.

Reply:
Yeah, I see that it should work as expected with tracing, because the `used` to create the tools lambda will have the proper context in place.